.buildkite/scripts/build_pr.sh

```diff
@@ -1,2 +1,97 @@
 #!/bin/bash
-echo "nothing to see yet"
+set -euo pipefail
+set +x
+
+# This script should only be invoked by the Buildkite PR bot
+if [ -z ${GITHUB_PR_BRANCH+set} ] || [ -z ${GITHUB_PR_TARGET_BRANCH+set} ] || [ -z ${GITHUB_PR_NUMBER+set} ] || [ -z ${GITHUB_PR_BASE_REPO+set} ];then
+  echo "One of the following env. variable GITHUB_PR_BRANCH, GITHUB_PR_TARGET_BRANCH, GITHUB_PR_NUMBER, GITHUB_PR_BASE_REPO is missing - exiting."
+  exit 1
+fi
+
+# Configure the git author and committer information
+export GIT_AUTHOR_NAME='Buildkite CI'
+export GIT_AUTHOR_EMAIL='docs-status+buildkite@elastic.co'
+export GIT_COMMITTER_NAME=$GIT_AUTHOR_NAME
+export GIT_COMMITTER_EMAIL=$GIT_AUTHOR_EMAIL
+
+# Set some metadata for build filtering capabilities
+# https://buildkite.com/elastic/docs-build-pr/builds?meta_data[repo_pr]=tech-content_123
+buildkite-agent meta-data set "repo_pr" "${GITHUB_PR_BASE_REPO}_${GITHUB_PR_NUMBER}"
+
+rebuild_opt=""
+build_args=""
+TARGET_BRANCH=""
+
+# Define build docs arguments
+if [[ ${GITHUB_PR_COMMENT_VAR_REBUILD_OPT:="unset"} == "rebuild" ]];then
+  rebuild_opt=" --rebuild"
+elif [[ ${GITHUB_PR_COMMENT_VAR_SKIP_OPT:="unset"} == "skiplinkcheck" ]];then
+  build_args+=" --skiplinkcheck"
+  if [[ ${GITHUB_PR_COMMENT_VAR_WARN_OPT:="unset"} == "warnlinkcheck" ]];then
```

Comment:
I haven't fully thought through how this works, but it doesn't seem like it's possible for `--warnlinkcheck` to get set unless `--skiplinkcheck` is also set, but I thought that `--skiplinkcheck` overrides `--warnlinkcheck`. Am I missing something?

Reply:
so theses flags will be passed by the Buildkite PR bot based on the comment left on the PR - as configured [here](https://github.com/elastic/docs/blob/master/.buildkite/pull-requests.org-wide.json#L14):

```
"trigger_comment_regex": "run docs-build ?(?<rebuild_opt>rebuild)? ?(?<warn_opt>warnlinkcheck)? ?(?<skip_opt>skiplinkcheck)?",
```

Reply:
What flags would you pass to get `--warnlinkcheck` to run? If you pass `GITHUB_PR_COMMENT_VAR_WARN_OPT=warnlink`, then `build_args` won't be modified unless you also pass `GITHUB_PR_COMMENT_VAR_SKIP_OPT=skiplinkcheck`. 

But you can't pass both, or the docs build [should raise an error](https://github.com/elastic/docs/blob/master/build_docs.pl#L1127-L1129).